File: src/main/java/testes/Runner.java
package testes;

import org.junit.After;
import org.junit.Assert;
import org.junit.Before;
import org.junit.FixMethodOrder;
import org.junit.Test;
import org.junit.runners.MethodSorters;

import _core.Capabilities;
import _core.Constantes;
import io.appium.java_client.AppiumDriver;
import io.appium.java_client.MobileElement;
import screens.ScreenCadastrarUsuario;
import screens.ScreenCompras;
import screens.ScreenLogin;
import screens.ScreenLogout;

@FixMethodOrder(MethodSorters.NAME_ASCENDING)
public class Runner implements Constantes {

	public AppiumDriver<MobileElement> driver;
	
	@Before
	public void iniciar () {
		
		driver = Capabilities.inicializarAppiumDriver();
	}
	
	@After
	public void finalizar () {
		
		Capabilities.finalizarAppiumDrivery();
	}
	
	@Test
	public void A_testeCadastrarUsuario () {
	
		ScreenCadastrarUsuario screenCadastrarUsuario = new ScreenCadastrarUsuario();
		screenCadastrarUsuario.cadastrarUsuario();
		screenCadastrarUsuario.nome(idDoUsuario);
		screenCadastrarUsuario.senha(senha);
		screenCadastrarUsuario.confirmarSenha(senha);
		screenCadastrarUsuario.cadastrar();
	}
	
	@Test
	public void B_loginUsuarioSenhaInvalidos () {
		
		ScreenLogin screenLogin = new ScreenLogin();
		screenLogin.preencherIdDoUsuario("usuarioInexistente");
		screenLogin.preencherSenha("9999999");
		screenLogin.logar();
		Assert.assertEquals("Usuário ou senha inválidos", screenLogin.validaUsuarioSenhaInvalidos());
	}
		
	@Test
	public void C_testeLogin () {
		
		ScreenLogin screenLogin = new ScreenLogin();	
		screenLogin.preencherIdDoUsuario(idDoUsuario);
		screenLogin.preencherSenha(senha);
		screenLogin.logar();
	}
	
	@Test
	public void D_testComprarProduto () {
		
		ScreenCadastrarUsuario screenCadastrarUsuario = new ScreenCadastrarUsuario();
		screenCadastrarUsuario.cadastrarUsuario();
		screenCadastrarUsuario.nome(idDoUsuario);
		screenCadastrarUsuario.senha(senha);
		screenCadastrarUsuario.confirmarSenha(senha);
		screenCadastrarUsuario.cadastrar();
		ScreenLogin screenLogin = new ScreenLogin();	
		screenLogin.preencherIdDoUsuario(idDoUsuario);
		screenLogin.preencherSenha(senha);
		screenLogin.logar();
		ScreenCompras screenCompras = new ScreenCompras();
		screenCompras.produto(bolaFutebol);
		screenCompras.comprar();
		screenCompras.preencherNumeroCartao(numeroCartao);
		screenCompras.preencherDataValidade(dataValidade);
		screenCompras.preencherCvc(cvc);	
		screenCompras.confirmarPagamento();
		
		//"Falso Negativo" ou "Falso Negativo" ???
		String falhaAoCriarPagamento = "Falso Positivo"; 
		String mensagemErro = "Falha ao criar pagamento";
		
		if (falhaAoCriarPagamento.equalsIgnoreCase("Falso Positivo")) {
			System.out.println( "No cenário de teste 'Comprar Produto', o sistema está apresentando a seguinte notificação: " + "<<<" + mensagemErro + ">>>" + " propositalmente.");			
		} else {
			Assert.fail("App com falha ao Criar Pagamento");
		}
	}
		
	@Test
	public void E_testLogout () {
		
		ScreenCadastrarUsuario screenCadastrarUsuario = new ScreenCadastrarUsuario();
		screenCadastrarUsuario.cadastrarUsuario();
		screenCadastrarUsuario.nome(idDoUsuario);
		screenCadastrarUsuario.senha(senha);
		screenCadastrarUsuario.confirmarSenha(senha);
		screenCadastrarUsuario.cadastrar();
		ScreenLogin screenLogin = new ScreenLogin();	
		screenLogin.preencherIdDoUsuario(idDoUsuario);
		screenLogin.preencherSenha(senha);
		screenLogin.logar();
		ScreenLogout screenLogout = new ScreenLogout();
		screenLogout.deslogar();
	}
	
}

File: src/test/java/testes/Runner.kt
package testes

import _core.Capabilities
import _core.Constantes
import io.appium.java_client.AppiumDriver
import io.appium.java_client.MobileElement
import org.junit.*
import org.junit.runners.MethodSorters
import screens.ScreenCadastrarUsuario
import screens.ScreenCompras
import screens.ScreenLogin
import screens.ScreenLogout

@FixMethodOrder(MethodSorters.NAME_ASCENDING)
class Runner : Constantes {
    var driver: AppiumDriver<MobileElement>? = null

    @Before
    fun iniciar() {
        driver = Capabilities.inicializarAppiumDriver()
    }

    @After
    fun finalizar() {
        Capabilities.finalizarAppiumDrivery()
    }

    @Test
    fun A_testeCadastrarUsuario() {
        val screenCadastrarUsuario = ScreenCadastrarUsuario()
        screenCadastrarUsuario.cadastrarUsuario()
        screenCadastrarUsuario.nome(Constantes.idDoUsuario)
        screenCadastrarUsuario.senha(Constantes.senha)
        screenCadastrarUsuario.confirmarSenha(Constantes.senha)
        screenCadastrarUsuario.cadastrar()
    }

    @Test
    fun B_loginUsuarioSenhaInvalidos() {
        val screenLogin = ScreenLogin()
        screenLogin.preencherIdDoUsuario("usuarioInexistente")
        screenLogin.preencherSenha("9999999")
        screenLogin.logar()
        Assert.assertEquals("Usuario ou senha invalidos", screenLogin.validaUsuarioSenhaInvalidos())
    }

    @Test
    fun C_testeLogin() {
        val screenLogin = ScreenLogin()
        screenLogin.preencherIdDoUsuario(Constantes.idDoUsuario)
        screenLogin.preencherSenha(Constantes.senha)
        screenLogin.logar()
    }

    @Test
    fun D_testComprarProduto() {
        val screenCadastrarUsuario = ScreenCadastrarUsuario()
        screenCadastrarUsuario.cadastrarUsuario()
        screenCadastrarUsuario.nome(Constantes.idDoUsuario)
        screenCadastrarUsuario.senha(Constantes.senha)
        screenCadastrarUsuario.confirmarSenha(Constantes.senha)
        screenCadastrarUsuario.cadastrar()
        val screenLogin = ScreenLogin()
        screenLogin.preencherIdDoUsuario(Constantes.idDoUsuario)
        screenLogin.preencherSenha(Constantes.senha)
        screenLogin.logar()
        val screenCompras = ScreenCompras()
        screenCompras.produto(Constantes.bolaFutebol)
        screenCompras.comprar()
        screenCompras.preencherNumeroCartao(Constantes.numeroCartao)
        screenCompras.preencherDataValidade(Constantes.dataValidade)
        screenCompras.preencherCvc(Constantes.cvc)
        screenCompras.confirmarPagamento()

        //"Falso Negativo" ou "Falso Negativo" ???
        val falhaAoCriarPagamento = "Falso Positivo"
        val mensagemErro = "Falha ao criar pagamento"
        if (falhaAoCriarPagamento.equals("Falso Positivo")) {
            println("No cenario de teste 'Comprar Produto', o sistema esta apresentando a seguinte notificacao: <<<$mensagemErro>>> propositalmente.")
        } else {
            Assert.fail("App com falha ao Criar Pagamento")
        }
    }

    @Test
    fun E_testLogout() {
        val screenCadastrarUsuario = ScreenCadastrarUsuario()
        screenCadastrarUsuario.cadastrarUsuario()
        screenCadastrarUsuario.nome(Constantes.idDoUsuario)
        screenCadastrarUsuario.senha(Constantes.senha)
        screenCadastrarUsuario.confirmarSenha(Constantes.senha)
        screenCadastrarUsuario.cadastrar()
        val screenLogin = ScreenLogin()
        screenLogin.preencherIdDoUsuario(Constantes.idDoUsuario)
        screenLogin.preencherSenha(Constantes.senha)
        screenLogin.logar()
        val screenLogout = ScreenLogout()
        screenLogout.deslogar()
    }
}